core/logger/logger.go

```diff
@@ -119,36 +129,31 @@ func (l *zapLogger) withCallerSkip(skip int) Logger {
 
 func (l *zapLogger) WarnIf(err error) {
 	if err != nil {
-		l.Warn(err)
+		l.withCallerSkip(1).Warn(err)
```

Comment:
These conditional helpers need to skip another level as well. We could precreate and store a skip logger on the struct instead of calling `withCallerSkip()` each time.